### ISSUE TITLE
When training the model, it takes a lot of time when the valid-steps are reached

### ISSUE THREAD

**Issue by eleven-0325** [OpenNMT/OpenNMT-py#2388]
When using the onmt 3.12
The following is the parameter settings：
                 -word_vec_size 512 \
                -hidden_size 512 \
                -num_workers 0   -bucket_size 1024 -model_dtype fp16 \
                -batch_size 1024 \
                -batch_size_multiple 4 \
                -valid_batch_size 1024 \
                -batch_type tokens -normalization tokens \
                -heads 8 \
                -transformer_ff 2048 \
                -min_length 20 -max_length 100 \
                -optim  adam  -decay_method noam  -max_grad_norm 0.0 -learning_rate 2 \
                -train_steps 20000  -report_every 500  -warmup_steps 1000 -valid_steps 1000\
                -train_metrics "BLEU" "TER"  -valid_metrics "BLEU" "TER" \

I reduced the validation set to 2500, and it took 20 minutes 
![image](https://github.com/OpenNMT/OpenNMT-py/assets/21076093/31ae9ad3-f734-476c-b5ec-079f83808439)

**Comment by vince62s** [comment]
first don't use a bucket of 1024 (128000 or more will way better)
second num_workers 1 or 2 will speed up things
last: 379 s is more like 6 mn in your screenshot, depends on the size of your valid set.

**Comment by eleven-0325** [comment]
Thank you very much, I will try it

**Comment by eleven-0325** [comment]
@vince62s 
Thanks for your reply, but it didn't improve after I tried   
When train the model, nvidia-smi
![image](https://github.com/OpenNMT/OpenNMT-py/assets/21076093/c593fb53-9b3c-42b6-92aa-e0bdcc7ac2fc)

but validate the dataset, nvidia-smi
![image](https://github.com/OpenNMT/OpenNMT-py/assets/21076093/eea1c7c8-8c72-49a5-b8cb-2eb99f5b9adf)
Only the first GPU is working, but the video memory does not change, and the occupancy and video memory of the other GPUs remain unchanged.


               -gpu_ranks 0 1 2 3 4 5 6 7 \
                --world_size 8 \
                -n_sample 0 \
                -encoder_type transformer -decoder_type transformer \
                -layers 6 \
                -word_vec_size 512 \
                -hidden_size 512 \
                -num_workers 2  -bucket_size 128000 -model_dtype fp16 \
                -batch_size 256 \
                -batch_size_multiple 100 \
                -valid_batch_size 51200 \
                -batch_type tokens -normalization tokens \
                -heads 8 \
                -transformer_ff 2048 \
                -min_length 20 -max_length 100 \

**Comment by eleven-0325** [comment]
valid stats calculation and sentences rebuilding
                           took: 4662.455519199371 s.
INFO:onmt:valid stats calculation and sentences rebuilding
                           took: 4662.455519199371 s.
 The translation of the valid dataset
                               took : 15027.757092952728 s.
INFO:onmt:The translation of the valid dataset
                               took : 15027.757092952728 s

**Comment by vince62s** [comment]
see #2360
